src/wp-includes/class-wp-user-query.php

```diff
@@ -804,28 +804,41 @@ public function query() {
 				{$this->query_orderby}
 				{$this->query_limit}
 			";
-
-			if ( is_array( $qv['fields'] ) ) {
-				$this->results = $wpdb->get_results( $this->request );
+			$cache_key   = $this->generate_cache_key( $qv, $this->request );
+			$cache_value = wp_cache_get( $cache_key, 'users' );
+			if ( false !== $cache_value ) {
```

Comment:
```suggestion
			if ( $cache_value ) {
```

Given the only return value we care about here is a properly formed array in the form of `[ 'user_data' => .., 'total_users' => ... ]` checking explicitly for `false` here doesn't seem warranted.

Reply:
What if it is an empty array?

Reply:
> What if it is an empty array?

If there is any risk of ambiguity, I'd use the `$found` parameter to avoid all doubt.

Reply:
I am not sure we need it here. In WP_Query it was used as the variable was needed in lots of places. Here is one simple check. If this is an array and empty, then something went seruosly wrong. Either it will be false and array with data.

Reply:
Using `$found` is more appropriate, but given the potential of caching empty arrays, I'm fine with your explanation :)